xof.go

```diff
@@ -0,0 +1,25 @@
+package kyber
+
+import (
+	"crypto/cipher"
+	"io"
+)
+
+// An XOF is an extendable output function, which is a cryptographic
+// primitive that can take arbitrary input like a hash, and then create
+// a stream of output, up to a limit determined by the size of the internal
+// state of the XOF.
+type XOF interface {
+	// Write absorbs more data into the hash's state. It panics if called
+	// after Read.
+	io.Writer
+
+	// Read reads more output from the hash. It returns io.EOF if the limit
+	// has been reached.
+	io.Reader
+
+	cipher.Stream
+
+	Clone() XOF
+	KeySize() int
```

Comment:
What is KeySize ? It seems to create a XOK you dont provide a key but a seed so what is the link ?

Reply:
I *think* it is the minimum amount of data that must be fed into to XOF to ensure that the full amount of output can be taken from it. For example, for blake, it is passed through from: https://godoc.org/golang.org/x/crypto/blake2b#BlockSize

But honestly, this is a part of the XOF that I don't feel I understand too well.

Reply:
From https://godoc.org/golang.org/x/crypto/blake2b#New256 it seems the key has no specific requirement on its length and since it returns a variable length hash function it's still a hash function and should not require any conditions on the input, no ? At least the doc don't mention anything in particular regarding the size of the input, or am I missing stg ?